tests/components/picnic/test_config_flow.py

```diff
@@ -122,3 +132,107 @@ async def test_form_exception(hass):
 
     assert result2["type"] == "form"
     assert result2["errors"] == {"base": "unknown"}
+
+
+async def test_form_already_configured(hass, picnic_api):
+    """Test that an entry with unique id can only be added once."""
+    # Create a mocked config entry and make sure to use the same user_id as set for the picnic_api mock response.
+    MockConfigEntry(
+        domain=DOMAIN,
+        unique_id=picnic_api().get_user()["user_id"],
+        data={CONF_ACCESS_TOKEN: "a3p98fsen.a39p3fap", CONF_COUNTRY_CODE: "NL"},
+    ).add_to_hass(hass)
+
+    result_init = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    result_configure = await hass.config_entries.flow.async_configure(
+        result_init["flow_id"],
+        {
+            "username": "test-username",
+            "password": "test-password",
+            "country_code": "NL",
+        },
+    )
+    await hass.async_block_till_done()
+
+    assert result_configure["type"] == data_entry_flow.RESULT_TYPE_ABORT
+    assert result_configure["reason"] == "already_configured"
+
+
+async def test_step_reauth(hass, picnic_api):
+    """Test the re-auth flow."""
+    # Create a mocked config entry
+    conf = {CONF_ACCESS_TOKEN: "a3p98fsen.a39p3fap", CONF_COUNTRY_CODE: "NL"}
+
+    MockConfigEntry(
+        domain=DOMAIN,
+        unique_id=picnic_api().get_user()["user_id"],
+        data=conf,
+    ).add_to_hass(hass)
+
+    # Init a re-auth flow
+    result_init = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_REAUTH}, data=conf
+    )
+    assert result_init["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result_init["step_id"] == "reauth_confirm"
+
+    with patch(
+        "homeassistant.components.picnic.async_setup_entry",
+        return_value=True,
+    ):
+        result_configure = await hass.config_entries.flow.async_configure(
+            result_init["flow_id"],
+            {
+                "username": "test-username",
+                "password": "test-password",
+            },
+        )
+        await hass.async_block_till_done()
+
+    # Check that the returned flow has type abort because of successful re-authentication
+    assert result_configure["type"] == data_entry_flow.RESULT_TYPE_ABORT
+    assert result_configure["reason"] == "reauth_successful"
+
+    assert len(hass.config_entries.async_entries()) == 1
```

Comment:
I've changed the flow so the country code is also asked as re-authentication, maybe it can change if someone moves from one country to the other, so it's fine to ask it again.